src/test/scala/scraml/libs/CirceJsonSupportSpec.scala

```diff
@@ -385,6 +389,51 @@ class CirceJsonSupportSpec extends AnyFlatSpec with Matchers with SourceCodeForm
               |}""".stripMargin.stripTrailingSpaces
           )
         )
+        keybasediscriminator.source.companion.map(_.toString().stripTrailingSpaces) should be(
+          Some("""object KeyBaseDiscriminator {
+                 |  import io.circe.Decoder.Result
+                 |  import io.circe._
+                 |  implicit lazy val decoder: Decoder[KeyBaseDiscriminator] = new Decoder[KeyBaseDiscriminator] {
+                 |    override def apply(c: HCursor): Result[KeyBaseDiscriminator] = c.value.asObject.toRight(DecodingFailure("expected object", c.history)).flatMap { obj =>
+                 |      val keySet = obj.keys.toSet
```

Comment:
since we have an `JsonObject` already, couldnt we just use `obj.contains` instead of `keySet.contains`?